src/Symfony/Component/Messenger/EventListener/ResetMemoryUsageListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageReceivedEvent;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+
+/**
+ * @author Tim Düsterhus <tim@tideways-gmbh.com>
+ */
+final class ResetMemoryUsageListener implements EventSubscriberInterface
```

Comment:
wdyt to add some debug/info log here ?
can it be usefull for ppl?

Reply:
got it, i was mostly commenting to [this](https://github.com/symfony/symfony/issues/59788#issuecomment-2668917134) thx

Reply:
so IIUC this middleware is not a good default to have? who will discover this if needed?

Reply:
The difference is that `gc_collect_cycles` is potentially CPU intensive (depending on how the assigned memory looks like). On the other hand, `reset_peak_memory_usage()` is not a costly operation.

So making the usage of `gc_collect_cycles` configurable allows projects to make different choices regarding performance (trading CPU load vs more aggressive memory reduction), which is why we are doing it in 7.3 (and why we  reverted the non-configurable usage introduced before)